src/app/code/community/Omise/Gateway/Model/Payment/Offsitepaynow.php

```diff
@@ -0,0 +1,162 @@
+
+    /**
+     * Instantiate state and set it to state object
+     *
+     * @param string        $payment_action
+     * @param Varien_Object $state_object
+     */
+    public function initialize($payment_action, $state_object) {
+        $payment = $this->getInfoInstance();
+        $order   = $payment->getOrder();
+        $invoice = $order->prepareInvoice()->register();
+        $charge = $this->processPayment($payment, $invoice->getBaseGrandTotal());
+        $payment->setCreatedInvoice($invoice)
+            ->setIsTransactionClosed(false)
+            ->setIsTransactionPending(true)
+            ->addTransaction(
+                Mage_Sales_Model_Order_Payment_Transaction::TYPE_CAPTURE,
+                $invoice,
+                false,
+                Mage::helper('omise_gateway')->__('Capturing an amount of %s via '.$this->getPaymentTitle().'.', $order->getBaseCurrency()->formatTxt($invoice->getBaseGrandTotal()))
+            );
+        $order->addRelatedObject($invoice);
+        if ($charge->isAwaitPayment() || $charge->isAwaitCapture() || $charge->isSuccessful()) {
+            $state_object->setState(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT);
+            $state_object->setStatus($order->getConfig()->getStateDefaultStatus(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT));
+            if($charge->isAwaitPayment() ) {
+                $state_object->setIsNotified(true);
+                $this->sendOrderConfirmationEmail($order, $charge);
+                $payment->setIsTransactionPending(true);
+                Mage::getSingleton('checkout/session')->setOmiseAuthorizeUri(Mage::getUrl('omise/checkout_paynow', array()));
+            }
+            
+            return;
+        }
+        $this->_suspectToBeFailed($payment);
+    }
+
+    /**
+     * @param Varien_Object $payment
+     * @param float $amount 
+     *
+     * @return Omise_Gateway_Model_Api_Charge
+     * @throws Mage_Core_Exception
+     */
+    public function processPayment(Varien_Object $payment, $amount) {
+        $order = $payment->getOrder();
+
+        return parent::_process(
+            $payment,
+            array(
+                'amount'      => $this->getAmountInSubunits($amount, $order->getBaseCurrencyCode()),
+                'currency'    => $order->getBaseCurrencyCode(),
+                'description' => 'Processing payment with '.$this->getPaymentTitle().'. Magento order ID: ' . $order->getIncrementId(),
+                'source'      => array('type' => $this->_type),
+                'return_uri'  => $this->getCallbackUri(),
+                'metadata'    => array(
+                    'order_id' => $order->getIncrementId()
+                )
+            )
+        );
+    }
+
+    /**
+     * Send Tesco Lotus Payment Barcode to customer.
+     * @param Mage_Sales_Model_Order $order
+     * @param Omise_Gateway_Model_Api_Charge $charge
+     * @return void
+     */
+    protected function sendOrderConfirmationEmail($order, $charge) {
```

Comment:
Function is all but identical to the one in Tesco payment method. This adds more weight to the above comment